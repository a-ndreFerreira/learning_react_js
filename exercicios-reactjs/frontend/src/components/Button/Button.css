.buttonClass {
    max-width: 30rem;
    margin: 0 auto;
    width: 100%;
    padding: 1rem 1.5rem;
    background-color: darkblue;
    color: #fff;
    cursor: pointer;
    transition: opacity .2s ease;
    opacity: .8;
}

.buttonClass:hover {
    opacity: 1;
}

.buttonClass:disabled {
    background-color: #222;
    cursor: not-allowed;
}