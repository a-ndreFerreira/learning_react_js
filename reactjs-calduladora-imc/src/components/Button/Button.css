button {
    text-transform: uppercase;
    padding: 1rem 1.5rem;
    background-color: #f44236;
    color: #fff;
    border: none;
    cursor: pointer;
    opacity: 0.8;
    flex: 1;
}

button:hover {
    opacity: 1;
}

#clearBtn {
    background-color: #444;
}