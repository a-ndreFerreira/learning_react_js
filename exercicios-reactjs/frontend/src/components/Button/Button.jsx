import './Button.css'

const Button = ({ text, onClick, disabled }) => {
    return (
        <button disabled={disabled} onClick={onClick} className='buttonClass'>
            {text}
        </button>
    )
}

export default Button