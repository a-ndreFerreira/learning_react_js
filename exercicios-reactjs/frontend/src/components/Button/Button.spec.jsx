import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { vi } from "vitest";
import { describe, expect } from "vitest";
import Button from "./Button";

describe('<Button />', () => {
    it('should render the button with the text', () => {
        render(<Button text='Load More Posts' />);
        expect.assertions(1);

        const button = screen.getByRole('button', { name: /Load More Posts/i })

        // expect(button).toHaveAttribute('class', 'buttonClass');
        expect(button).toBeInTheDocument();
    })

    it('should call function on button click', async () => {

        //funcao mock do vitest
        const fn = vi.fn();
        render(<Button text='Load More Posts' onClick={fn} />);

        const button = screen.getByRole('button', { name: /Load More Posts/i });

        await userEvent.click(button);
        // fireEvent.click(button);

        //segundo click
        // fireEvent.click(button);

        //se foi chamada, independente das vezes
        // expect(fn).toHaveBeenCalled();

        //se foi chamada pelo numero de vezes setado
        expect(fn).toHaveBeenCalledTimes(1);

    })

    it('should be enabled when disabled is false', () => {
        render(<Button text='Load More Posts' disabled={false} />);

        const button = screen.getByRole('button', { name: /Load More Posts/i });

        //toBeDisabeled para ver se esta desativado, e toBeEnabled para ver se esta ativado
        expect(button).toBeEnabled();

    })
})