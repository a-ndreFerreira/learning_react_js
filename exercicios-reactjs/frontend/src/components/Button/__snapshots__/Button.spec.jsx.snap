// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot 1`] = `
<button
  class="buttonClass"
>
  Load More Posts
</button>
`;
